src/databricks/labs/remorph/helpers/recon_config_utils.py

```diff
@@ -0,0 +1,186 @@
+import json
+import os
+
+from databricks.connect import DatabricksSession
+from databricks.labs.blueprint.entrypoint import get_logger
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+
+from databricks.labs.remorph.helpers.db_workspace_utils import DBWorkspaceClient
+from databricks.labs.remorph.reconcile.connectors.snowflake import SnowflakeDataSource
+from databricks.labs.remorph.reconcile.constants import SourceType
+from databricks.labs.remorph.reconcile.recon_config import TableRecon
+
+logger = get_logger(__file__)
+
+recon_source_choices = [
+    SourceType.SNOWFLAKE.value,
+    SourceType.ORACLE.value,
+    SourceType.DATABRICKS.value,
+    SourceType.NETEZZA.value,
+]
+
+
+class ReconConfigPrompts:
+    def __init__(self, ws: WorkspaceClient):
+        self._source = None
+        self._prompts = Prompts()
+        self._db_ws = DBWorkspaceClient(ws, self._prompts)
+
+    def prompt_source(self):
+        source = self._prompts.choice("Select the source", recon_source_choices)
+        self._source = source
+        return source
+
+    def prompt_snowflake_connection_details(self) -> tuple[str, dict[str, str]]:
+        logger.info(
+            f"Please answer a couple of questions to configure `{SourceType.SNOWFLAKE.value}` Connection profile"
+        )
+
+        sf_url = self._prompts.question("Enter Snowflake URL")
+        account = self._prompts.question("Enter Account Name")
+        sf_user = self._prompts.question("Enter User")
+        sf_password = self._prompts.question("Enter Password")
+        sf_db = self._prompts.question("Enter Database")
+        sf_schema = self._prompts.question("Enter Schema")
+        sf_warehouse = self._prompts.question("Enter Snowflake Warehouse")
+        sf_role = self._prompts.question("Enter Role", default=" ")
+
+        sf_conn_details = [
```

Comment:
Why not defining this directly as a dict?

Reply:
Done